app/Models/Presenters/EntityPresenter.php

```diff
@@ -84,14 +84,14 @@ public function getCityState()
         }
     }
 
-    public function getShippingCityState()
+    public function getShippingCityState($printCity = true, $printState = true, $printPostalCode = true)
     {
         $client = $this->entity;
         $swap = $client->shipping_country && $client->shipping_country->swap_postal_code;
 
-        $city = e($client->shipping_city);
-        $state = e($client->shipping_state);
-        $postalCode = e($client->shipping_postal_code);
+        $city = ($printCity) ? e($client->shipping_city) : null;
+        $state = ($printState) ? e($client->shipping_state) : null;
+        $postalCode = ($printPostalCode) ? e($client->shipping_postal_code) : null;
```

Comment:
This will always return the client->shipping information, sometime the data that is passed back could be from another source such as the $this->settings object.

As such, instead of modifying the existing method, i would suggest creating a new method just for this particular use case.

Reply:
I didnt' find any reference of this function. Is it in use?
My approach depends on "this function is not used right now" and when it should be used somewhere (in the future), the call can be modified with the variables of the function... In the end it's calling ```cityStateZip``` with the parameters from line 92-94 which can be null and therefore the address can populate the country specifics. Am I missing something?

Reply:
oh it is being used throughout the application. just search for cityStateZip(

also, in your code you are forcing the Client $client type when you cast

$client = $this->entity

The object _may_ be a client / vendor or even a settings object.

If you search for where we use this function, you'll see the possible types are being passed in.

Reply:
> oh it is being used throughout the application. just search for cityStateZip(

I think we are a bit off. I'm talking about ```getShippingCityState(``` where lines 92-94 are coming from. I only added the possibility to leave things out.

The ```cityStateZip(``` is modified as discussed earlier and I tried to check all references throughout the application. This function is not touched at all, but I added ```postal_city``` all over the place to add the desired variant of billing address.